### ISSUE TITLE
ImageNet-100 is not able to finetune

### ISSUE THREAD

**Issue by Junjie31** [vimar-gu/MinimaxDiffusion#5]
After I prepare the complete ImageNet dataset and the corresponding checkpoints, woof categories can achieve close to paper accuracy, but when I expand the classes and use ImageNet-100 in the finetune phase, the images generated by DiT are completely unavailable(quite a mess), is there some wrong parameter I'm using? Or are there other considerations when there are more categories?

```shell
torchrun --nnode=1 --master_port=25623 train_dit.py --model DiT-XL/2 \
     --data-path ./imagenet/train/ --ckpt ./DiT-XL-2-256x256.pt \
     --global-batch-size 8 --tag minimax --ckpt-every 12000 --log-every 1500 --epochs 8 \
     --condense --finetune-ipc -1 --results-dir ./logs/run1 --spec 100 --nclass 100 \
```

**Comment by vimar-gu** [comment]
Thanks for the interest in this work! 

As the computational resources in our lab was quite limited, I cannot increase the mini-batch size for the fine-tuning process. You can notice that the mini-batch size used is quite small. The current hyper-parameter setting cannot be directly adapted to circumstances with many classes. 

In our experiments, we found that the adopted hyper-parameters are suitable for 10/20 classes. So in the ImageNet-100/ImageNet-1K experiments, I separated the whole process into multiple phases. There is a `--phase` argument which is used to control the fine-tuning/generating phases. Empirically using 10 or 20 classes for one phase both yield good results. 

Please try the strategy. And of course, if there is sufficient resources, you can also try tuning the hyper-parameters for fine-tuning more classes at one time.

**Comment by Junjie31** [comment]
It turns out to be finetune by separating the dataset by phase, very COOL and effective idea, thanks a lot!

**Comment by yiqiqiandefanhua** [comment]
> It turns out to be finetune by separating the dataset by phase, very COOL and effective idea, thanks a lot!
Hi
How do you fine-tune the code in stages, can you share the code, the images generated by the model I fine-tuned are also very messy
thanks
